tests/test_filters.py

```diff
@@ -905,3 +905,35 @@ def test_poly_data_strip():
     slc = mesh.slice(normal='z', origin=(0,0,-10))
     stripped = slc.strip()
     assert stripped.n_cells == 1
+
+
+def test_shrink():
+
+    # mesh points
+    vertices = np.array([[0, 0, 0],
+                         [1, 0, 0],
+                         [1, 1, 0],
+                         [0, 1, 0],
+                         [0.5, 0.5, -1]])
+
+    # mesh faces
+    faces = [[4, 0, 1, 2, 3],  # square
+             [3, 0, 1, 4],     # triangle
+             [3, 1, 2, 4]]     # triangle
+
+    # number of points after shrink
+    n_points = 0
+    for face in faces:
+        n_points += face[0]
+
+    mesh = pyvista.PolyData(vertices, np.hstack(faces))
+    shrunk = mesh.shrink(shrink_factor=0.8)
+    assert shrunk.n_points == n_points
+    # assert shrunk.area < mesh.area
+
+
+    # Cube
+    mesh = pyvista.Cube()
+    shrunk = mesh.shrink(shrink_factor=0.8)
+    assert shrunk.n_points == mesh.n_points
```

Comment:
I am wondering that PolyData has different number of points after shrink. Some meshes have the same number of points after they are shrunk as before. The other meshes, on the other hand, exist independently for the number of points or each cell after the shrink. Is this a PyVista problem or VTK problem?

Reply:
I guess we can't make the assumption that there's the same number of points.  We should just assert that the number of faces is the same before and after.

Reply:
In most cases, the number of points in the shrink mesh will be greater than the original. For example, when shrinking a sphere, the shrunk mesh will easily have more points than the original because each cell must be shrunk and thus copies must be made of previously shared vertices to construct the now disconnected cells.

| original | shrunk |
|---|---|
| 50 points | 288 points |
| 96 cells | 96 cells |
| <img width="920" alt="Screen Shot 2020-10-02 at 8 55 23 AM" src="https://user-images.githubusercontent.com/22067021/94937609-0a060a80-048d-11eb-9f4d-9b52acce74ec.png"> | <img width="918" alt="Screen Shot 2020-10-02 at 8 55 32 AM" src="https://user-images.githubusercontent.com/22067021/94937626-0f635500-048d-11eb-8a8e-9d5885d44db1.png"> |

And to prove my point: 96 * 3 = 288 (3 points per triangle cell and there are no shared points)

Reply:
Thanks @akaszynski @banesullivan How about using this assert?
```Python
    assert shrunk.n_points <= mesh.n_points
```

Reply:
If the original mesh already has isolated faces, then this won't work either.  Let's just assert that we have the same number of faces, and if it's a `PolyData` that the area is the same.

Reply:
The fact that the area is the same makes me very, very uncomfortable. They should be different, right?! Look at my example above, a significant portion of the mesh is removed and thus the surface area should decrease.

Yet, both the `area` property and the `compute_cell_sizes` filter are showing ann equal area.

Reply:
But the areas are not equal....

```py
def get_area(mesh):
    return sum(mesh.compute_cell_sizes(length=False, area=True, volume=False,)["Area"])

import pyvista as pv

mesh = pv.Cube()
shrunk = mesh.shrink(0.8)
```
```py
>>> get_area(mesh), get_area(shrunk)
(6.0, 3.840000114440919)
```

Also. there must be a bug with `vtkMassProperties` as `shrunk.extract_surface().area` return `0.0`

Reply:
Let's discuss it in #926 .